autoware_reference_system/test/generate_traces.py

```diff
@@ -29,50 +28,63 @@
 
 # this file has @variables@ that are meant to be automatically replaced
 # by values using the `configure_file` CMake function during the build
+RUNTIME = int('@RUNTIME@')
 
 
 def generate_test_description():
-    env = os.environ.copy()
-    env['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = '[{severity}] [{name}]: {message}'
-
     # replaced with cmake `configure_file` function
     rmw_impl = '@RMW_IMPLEMENTATION@'
     test_exe = '@TEST_EXECUTABLE@'
     test_exe_name = '@TEST_EXECUTABLE_NAME@'
     timeout = int('@TIMEOUT@')
-    extra_timeout = timeout + 5
-
-    # specify rmw to use
-    env['RCL_ASSERT_RMW_ID_MATCHES'] = rmw_impl
-    env['RMW_IMPLEMENTATION'] = rmw_impl
 
     launch_description = LaunchDescription()
+
+    # see https://github.com/ros2/launch/issues/417
+    # have to use `SetEnvironmentVariable` to set env vars in launch description
+    envvar_rcutils_action = SetEnvironmentVariable(
+        'RCUTILS_CONSOLE_OUTPUT_FORMAT', '[{severity}] [{name}]: {message}')
+    envvar_rmw_action = SetEnvironmentVariable(
+        'RMW_IMPLEMENTATION', rmw_impl)
+    envvar_rclassert_rmw_action = SetEnvironmentVariable(
+        'RCL_ASSERT_RMW_ID_MATCHES', rmw_impl)
+
     proc_under_test = ExecuteProcess(
         cmd=[test_exe],
         name=test_exe_name,
-        sigterm_timeout=str(extra_timeout),
-        output='screen',
-        env=env,
+        sigterm_timeout=str(timeout),
+        output='screen'
     )
```

Comment:
you should probably use `Node` instead of `ExecuteProcess`. This way you don't even need to know the actual path to the executable (and can remove `test_exe` from here and the `CMakeLists.txt`). You just need the package name and the executable name. Example: https://gitlab.com/ros-tracing/ros2_tracing/-/blob/ad79dad92013e554b902bfcb460003038f14a8a3/tracetools_launch/launch/example.launch.py#L27-L31